src/Symfony/Component/Lock/Store/DbalStore.php

```diff
@@ -0,0 +1,286 @@
+        $table->setPrimaryKey([$this->idCol]);
+
+        foreach ($schema->toSql($conn->getDatabasePlatform()) as $sql) {
+            $conn->executeStatement($sql);
+        }
+    }
+
+    /**
+     * Cleans up the table by removing all expired locks.
+     */
+    private function prune(): void
+    {
+        $sql = "DELETE FROM $this->table WHERE $this->expirationCol <= {$this->getCurrentTimestampStatement()}";
+
+        $conn = $this->getConnection();
+        $conn->executeStatement($sql);
+    }
+
+    private function getDriver(): string
+    {
+        if (null !== $this->driver) {
+            return $this->driver;
+        }
+
+        $driver = $this->getConnection()->getDriver();
+        $platform = $driver->getDatabasePlatform();
+
+        if ($driver instanceof \Doctrine\DBAL\Driver\Mysqli\Driver) {
+            throw new \LogicException(sprintf('The adapter "%s" does not support the mysqli driver, use pdo_mysql instead.', static::class));
+        }
+
+        switch (true) {
+            case $platform instanceof \Doctrine\DBAL\Platforms\MySQLPlatform:
+            case $platform instanceof \Doctrine\DBAL\Platforms\MySQL57Platform:
+                $this->driver = 'mysql';
+                break;
+            case $platform instanceof \Doctrine\DBAL\Platforms\SqlitePlatform:
+                $this->driver = 'sqlite';
+                break;
+            case $platform instanceof \Doctrine\DBAL\Platforms\PostgreSQLPlatform:
+            case $platform instanceof \Doctrine\DBAL\Platforms\PostgreSQL94Platform:
+                $this->driver = 'pgsql';
+                break;
+            case $platform instanceof \Doctrine\DBAL\Platforms\OraclePlatform:
+                $this->driver = 'oci';
+                break;
+            case $platform instanceof \Doctrine\DBAL\Platforms\SQLServerPlatform:
+            case $platform instanceof \Doctrine\DBAL\Platforms\SQLServer2012Platform:
+                $this->driver = 'sqlsrv';
+                break;
+            default:
+                $this->driver = \get_class($platform);
+                break;
+        }
+
+        return $this->driver;
+    }
+
+    /**
+     * Provides an SQL function to get the current timestamp regarding the current connection's driver.
+     */
+    private function getCurrentTimestampStatement(): string
+    {
+        switch ($this->getDriver()) {
+            case 'mysql':
+                return 'UNIX_TIMESTAMP()';
+            case 'sqlite':
+                return 'strftime(\'%s\',\'now\')';
+            case 'pgsql':
+                return 'CAST(EXTRACT(epoch FROM NOW()) AS INT)';
+            case 'oci':
+                return '(SYSDATE - TO_DATE(\'19700101\',\'yyyymmdd\'))*86400 - TO_NUMBER(SUBSTR(TZ_OFFSET(sessiontimezone), 1, 3))*3600';
+            case 'sqlsrv':
+                return 'DATEDIFF(s, \'1970-01-01\', GETUTCDATE())';
+            default:
+                return time();
```

Comment:
```suggestion
                return (string) time();
```

Psalm is nit-picking here. 😅

Reply:
Fixed on `PdoStore` too.